netty/src/main/java/io/grpc/netty/GrpcHttp2HeadersUtils.java

```diff
@@ -418,8 +421,12 @@ public CharSequence scheme() {
     public List<CharSequence> getAll(CharSequence csName) {
       AsciiString name = requireAsciiString(csName);
       if (isPseudoHeader(name)) {
-        // This code should never be reached.
-        throw new IllegalArgumentException("Use direct accessor methods for pseudo headers.");
+        AsciiString value = getPseudoHeader(name);
+        if (value == null) {
+          return Collections.emptyList();
+        } else {
+          return Collections.singletonList(value);
+        }
```

Comment:
Not a fan of the conditional operator? :)

Reply:
Honestly, I didn't even consider it. I do use it, albeit less as time goes on. I'd say I don't use it any more, but I think I sometimes notice it is missing in Go, so I do sometimes use it. Here I have no desire, I think because it isn't trivial code. Like, neither case is a literal. But even then... dunno.

This particular is also interesting, because I didn't write:
```java
if (value == null) {
  return Collections.emptyList();
}
return Collections.singletonList(value);
```

But that's because I don't consider `value == null` an error. These two cases are symmetric and equal in complexity, so I have them mirror each other. If `value != null` case was 3 lines, I'd have changed the style. But again, I didn't even consider that as an option. I had a mental model and I wrote it down.